client/components/editor/teaching-elements/Assessment/TextSelection/Selector.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div
+    v-html="structure"
+    :class="{ editing: this.isEditing }"
+    @mouseup="e => isEditing && onMouseup(e)"
+    class="selector ql-editor">
+  </div>
+</template>
+
+<script>
+import {
+  generateStructure,
+  getSelectionRange,
+  getText,
+  isBlast,
+  toggleAttributes,
+  toggleClasses,
+  nodeMapper
+} from './helpers';
+import Selections from './Selections';
+
+export default {
+  props: {
+    correct: { type: Array, required: true },
+    isEditing: { type: Boolean, required: true },
+    content: { type: String, required: true }
+  },
+  data() {
+    const selections = new Selections(this.correct);
+    const processNode = selections.processNode.bind(selections);
+    return {
+      selections,
+      structure: generateStructure(this.content, (node, index) => {
+        return nodeMapper(processNode, node, index);
+      })
+    };
+  },
+  methods: {
+    onMouseup(event) {
+      const node = event.target;
+      const selection = getSelectionRange();
+      if (selection && !selection.isCollapsed) {
+        this.addSelection(selection);
+      } else if (Selections.isSelected(node) && isBlast(node)) {
+        const range = Selections.extractRange(node);
+        this.removeSelection(range);
```

Comment:
Huh, all is great here from control flow standpoint but in the same time:
- `getSelectionRange` returns `selection` (which is in fact range)
- `this.addSelection` accepts `selection`
- `Selections.extractRange` returns `range` (ok)
- `this.removeSelection` accepts `range` (should be `removeRange` then?)

Naming is inconsistent and instead of lowering cognitive load it actually increases it 😢